File: courses/lesson4_1recycle_view_contacts/src/main/java/by/academy/lesson4/part1/contacts/DataStorage.java
package by.academy.lesson4.part1.contacts;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

public class DataStorage implements Parcelable {

    private List<DataItem> items = new ArrayList<DataItem>() {{
        add(new DataItem("Ann", "+1111111"));
        add(new DataItem("Bob", "2222@tet.com"));
        add(new DataItem("Conrad", "+3333"));
        add(new DataItem("Dorry", "444@droff.org"));
        add(new DataItem("Enie", "+5555"));
        add(new DataItem("Frank", "66@pust.com"));
    }};

    public DataStorage() {
    }

    protected DataStorage(Parcel in) {
        items = in.createTypedArrayList(DataItem.CREATOR);
    }

    public static final Creator<DataStorage> CREATOR = new Creator<DataStorage>() {
        @Override
        public DataStorage createFromParcel(Parcel in) {
            return new DataStorage(in);
        }

        @Override
        public DataStorage[] newArray(int size) {
            return new DataStorage[size];
        }
    };

    public List<DataItem> getItems() {
        return items;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeList(items);
    }
}

File: courses/lesson5_cars/src/main/java/by/academy/lesson5/cars/DataStorage.kt
package by.academy.lesson5.cars

import android.os.Parcel
import android.os.Parcelable
import java.util.*

class DataStorage() : Parcelable {
    var items: List<DataItem?>? = object : ArrayList<DataItem?>() {
        init {
            add(DataItem("Ann", "Rolls Royce", "Phantom", "1112 HA-8"))
            add(DataItem("Bob", "Rolls Royce", "Phantom", "2222 HA-8"))
            add(DataItem("Conrad", "Ford", "Mustang", "3333 HA-8"))
            add(DataItem("Dorry", "Tesla", "Model Y", "4444 HA-8"))
//            add(DataItem("Ennio", "Toyota", "Corolla", "5555 HA-8"))
//            add(DataItem("Frank", "Volkswagen", "Golf", "6612 HA-8"))
        }
    }

    constructor(parcel: Parcel) : this() {
        items = parcel.createTypedArrayList(DataItem)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeList(items)
    }

    companion object CREATOR : Parcelable.Creator<DataStorage> {
        override fun createFromParcel(parcel: Parcel): DataStorage {
            return DataStorage(parcel)
        }

        override fun newArray(size: Int): Array<DataStorage?> {
            return arrayOfNulls(size)
        }
    }


}